src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -888,16 +913,17 @@ long nextInterval(long now) {
               boolean isWorkerPausedFromNewWork() {
                 try {
                   File pausedFile = new File(config.getRoot() + "/.paused");
```

Comment:
The handler script should be responsible for the deletion. I use lifecycled to listen for SQS termination events and run a script, which creates the file and monitors the execution slots used metric to determine when all work has been completed. When all clear, the script ends and lifecycled sends a message to the queue to terminate the instance. The script can also delete the file if needed. I don't because I'm expecting the host to terminate.